src/net_processing.cpp

```diff
@@ -5729,6 +5918,16 @@ bool PeerManagerImpl::SendMessages(CNode* pto)
                         if (tx_relay->m_bloom_filter) {
                             if (!tx_relay->m_bloom_filter->IsRelevantAndUpdate(*txinfo.tx)) continue;
                         }
+                        if (WITH_LOCK(m_mempool.cs, return m_mempool.IsUnbroadcastTx(GenTxid::Txid(txinfo.tx->GetHash())))) {
+                            LogPrintLevel( /* Continued */
+                                BCLog::SENSITIVE_RELAY,
+                                BCLog::Level::Info,
+                                "Omitting INV for unbroadcast transaction (txid=%s) from MEMPOOL reply peer=%d%s\n",
+                                txinfo.tx->GetHash().ToString(),
+                                pto->GetId(),
+                                fLogIPs ? strprintf(", peeraddr=%s", pto->addr.ToStringAddrPort()) : "");
+                            continue;
+                        }
```

Comment:
This, together with the other change from commit 0ec7b979e7a9d4c30a97c39c2a64768d7c5662b1 `net_processing: omit unbroadcast txs from replies to GETDATA and MEMPOOL` alter the behavior even if sensitive relay is not used (e.g. disabled or Tor and I2P not reachable).

I think it is beneficial in that case too, but is not the purpose of this PR to improve that. Should this be guarded by `if (UseSensitiveRelay())`?

Reply:
For `GETDATA`, this only applies to transactions which we have not `INV`ed to the peer. Otherwise `INV`ed transactions will be sent via:

https://github.com/bitcoin/bitcoin/blob/f4a8269dfc144cc918570bdb870aa5143a11c1fe/src/net_processing.cpp#L2307